### PR TITLE
Adds to `test_requires`

### PR DESCRIPTION
**Pull request by prabhatnagarajan**
The tests use `scipy` and `optuna`, so I include them in the `test_requires`. I have linked in comments to the places where they are used in the tests.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
     'gym>=0.9.7',
-    'numpy>=1.10.4',
+    'numpy>=1.11.0',
     'pillow',
     'filelock',
 ]
 
 test_requires = [
     'pytest',
+    'scipy',
```
Comment (prabhatnagarajan) [inline setup.py:15]: The tests use it.
https://github.com/pfnet/pfrl/blob/b29533b77c82f88fbc074714274ee8aa63dc270f/tests/utils_tests/test_random.py#L6

File: setup.py [hunk 2]
```diff
@@ -5,13 +5,15 @@
...
+    'optuna',
```
Comment (prabhatnagarajan) [inline setup.py:16]: The tests use it.
https://github.com/pfnet/pfrl/blob/b29533b77c82f88fbc074714274ee8aa63dc270f/tests/experiments_tests/test_evaluation_hooks.py#L4

File: setup.py [hunk 3]
```diff
@@ -5,13 +5,15 @@
 install_requires = [
     'torch>=1.3.0',
     'gym>=0.9.7',
-    'numpy>=1.10.4',
+    'numpy>=1.11.0',
```
Comment (prabhatnagarajan) [inline setup.py:8]:
```suggestion
    'numpy>=1.10.4',
```

**Review by muupan — APPROVED** [review]
Thanks for the fix! It seems like the CI errors are not due to this PR.

**Pull request merged**